pkg/hubble/parser/seven/http.go

```diff
@@ -80,3 +84,36 @@ func (p *Parser) httpSummary(flowType accesslog.FlowType, http *accesslog.LogRec
 	}
 	return ""
 }
+
+// filterHeader receives a key-value pair of an http header along with an HttpHeadersList.
+// Based on the allow/deny lists of the provided HttpHeadersList it returns the original value
+// or the redacted constant "HUBBLE_REDACTED" accordingly:
+//  1. If both allow/deny lists are empty but hubble.redact feature is enabled the value of the
+//     header is redacted by default.
+//  2. If an allow list is provided, then if the header's key is contained in the allow list the value
+//     of the header will not be redacted.
+//  3. If a deny list is provided, then if the header's key is contained in the deny list the value
+//     of the header will be redacted.
+//  4. If none of the above happens, then if there is any allow list defined then the value will be redacted
+//     otherwise if there is a deny list defined the value will not be redacted.
```

Comment:
```suggestion
//  1. If `HttpHeadersList` is not nil (meaning that hubble.redact feature is enabled) but both allow/deny lists are empty then the value of the
//     header is redacted by default.
//  2. If the header's key is contained in the allow list then the value
//     of the header will not be redacted.
//  3. If the header's key is contained in the deny list then the value
//     of the header will be redacted.
//  4. If none of the above happens, then if there is any allow list defined then the value will be redacted
//     otherwise if there is a deny list defined the value will not be redacted.
```

I would change the wording a bit to avoid double ifs in points 2 and 3 and clarify how it's checked that hubble.redact is enabled.